libraries/AP_Landing/AP_Landing_Deepstall.cpp

```diff
@@ -0,0 +1,380 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ *   AP_Landing_Deepstall.cpp - Landing logic handler for ArduPlane for deepstall landings
+ */
+
+#include "AP_Landing.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_HAL/AP_HAL.h>
+#include <SRV_Channel/SRV_Channel.h>
+
+void AP_Landing::type_deepstall_do_land(const AP_Mission::Mission_Command& cmd, const float relative_altitude)
+{
+    type_deepstall_stage = DEEPSTALL_STAGE_FLY_TO_LANDING;
+    type_deepstall_PID.reset_I();
+
+    // load the landing point in, the rest of path building is deferred for a better wind estimate
+    memcpy(&type_deepstall_landing_point, &cmd.content.location, sizeof(Location));
+}
+
+// currently identical to the slope aborts
+void AP_Landing::type_deepstall_verify_abort_landing(const Location &prev_WP_loc, Location &next_WP_loc, bool &throttle_suppressed)
+{
+    // when aborting a landing, mimic the verify_takeoff with steering hold. Once
+    // the altitude has been reached, restart the landing sequence
+    throttle_suppressed = false;
+    nav_controller->update_heading_hold(get_bearing_cd(prev_WP_loc, next_WP_loc));
+}
+
+
+/*
+  update navigation for landing
+ */
+bool AP_Landing::type_deepstall_verify_land(const Location &prev_WP_loc, Location &next_WP_loc, const Location &current_loc,
+        const float height, const float sink_rate, const float wp_proportion, const uint32_t last_flying_ms,
+        const bool is_armed, const bool is_flying, const bool rangefinder_state_in_range)
+{
+    Location entry_point;
+    SRV_Channel* elevator;
+    float travel_distance;
+
+    switch (type_deepstall_stage) {
+    case DEEPSTALL_STAGE_FLY_TO_LANDING:
+        if (get_distance(current_loc, type_deepstall_landing_point) > 2 * aparm.loiter_radius) {
+            nav_controller->update_waypoint(current_loc, type_deepstall_landing_point);
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_ESTIMATE_WIND;
+        type_deepstall_loiter_sum_cd = 0; // reset the loiter counter
+        //fallthrough
```

Comment:
Michael, I agree with you. I've only every heard it called fall through. I only brought it up because the was some automated code system that would throw a warning unless you had a special name there. Can we have 2 comments?